src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -240,4 +248,131 @@ public void unarchiveChannel(Long channelId) {
         conversationService.notifyAllConversationMembersAboutUpdate(updatedChannel);
     }
 
+    /**
+     * Create a channel for a lecture
+     *
+     * @param lecture     the lecture to create the channel for
+     * @param channelName the name of the channel
+     * @return the created channel
+     */
+    public Channel createLectureChannel(Lecture lecture, @NotNull String channelName) {
+        Channel channelToCreate = new Channel();
+        channelToCreate.setName(channelName);
+        channelToCreate.setIsPublic(true);
+        channelToCreate.setIsAnnouncementChannel(false);
+        channelToCreate.setIsArchived(false);
+        channelToCreate.setDescription("Channel for lecture: " + lecture.getTitle());
+        channelToCreate.setLecture(lecture);
+        return createChannel(lecture.getCourse(), channelToCreate, Optional.of(userRepository.getUserWithGroupsAndAuthorities()));
+    }
+
+    /**
+     * Creates a channel for a course exercise and sets the channel name of the exercise accordingly
+     *
+     * @param exercise    the exercise to create the channel for
+     * @param channelName the name of the channel
+     * @return the created channel
+     */
+    public Channel createExerciseChannel(Exercise exercise, @NotNull String channelName) {
+        if (!exercise.isCourseExercise()) {
+            return null;
+        }
+
+        Channel channelToCreate = new Channel();
+        channelToCreate.setName(channelName);
+        channelToCreate.setIsPublic(true);
+        channelToCreate.setIsAnnouncementChannel(false);
+        channelToCreate.setIsArchived(false);
+        channelToCreate.setDescription(String.format("Channel for %s exercise: %s", exercise.getExerciseType().getExerciseTypeAsReadableString(), exercise.getTitle()));
```

Comment:
Idea for a potential followup: Don't use a fixed string here but generate this in the client. Would allow for
- translating this
- adding an additional description while keeping the linking
- adding some direct links for quickly navigating between channel and exercise.

Reply:
We will keep this in mind for a follow up, if needed. Good idea.